tests/nativeapp/test_event_sharing.py

```diff
@@ -0,0 +1,1086 @@
+    mock_connection,
+)
+from tests.nativeapp.utils import (
+    APP_ENTITY_GET_EXISTING_APP_INFO,
+    GET_UI_PARAMETERS,
+    SQL_EXECUTOR_EXECUTE,
+    mock_execute_helper,
+)
+from tests.testing_utils.fixtures import MockConnectionCtx
+
+allow_always_policy = AllowAlwaysPolicy()
+ask_always_policy = AskAlwaysPolicy()
+deny_always_policy = DenyAlwaysPolicy()
+test_manifest_contents = dedent(
+    """\
+    manifest_version: 1
+
+    version:
+        name: dev
+        label: "Dev Version"
+        comment: "Default version used for development. Override for actual deployment."
+
+    artifacts:
+        setup_script: setup.sql
+        readme: README.md
+
+    configuration:
+        log_level: INFO
+        trace_level: ALWAYS
+"""
+)
+
+test_manifest_with_mandatory_events = dedent(
+    """\
+    manifest_version: 1
+
+    version:
+        name: dev
+        label: "Dev Version"
+        comment: "Default version used for development. Override for actual deployment."
+
+    artifacts:
+        setup_script: setup.sql
+        readme: README.md
+
+    configuration:
+        telemetry_event_definitions:
+          - type: ERRORS_AND_WARNINGS
+            sharing: MANDATORY
+          - type: DEBUG_LOGS
+            sharing: OPTIONAL
+"""
+)
+
+
+def _create_or_upgrade_app(
+    policy: PolicyBase,
+    install_method: SameAccountInstallMethod,
+    is_interactive: bool = False,
+    package_id: str = "app_pkg",
+    app_id: str = "myapp",
+    console: AbstractConsole | None = None,
+):
+    dm = DefinitionManager()
+    pd = dm.project_definition
+    pkg_model: ApplicationPackageEntityModel = pd.entities[package_id]
+    app_model: ApplicationEntityModel = pd.entities[app_id]
+    ctx = WorkspaceContext(
+        console=console or cc,
+        project_root=dm.project_root,
+        get_default_role=lambda: "mock_role",
+        get_default_warehouse=lambda: "mock_warehouse",
+    )
+    app = ApplicationEntity(app_model, ctx)
+    pkg = ApplicationPackageEntity(pkg_model, ctx)
+    stage_fqn = f"{pkg_model.fqn.name}.{pkg_model.stage}"
+
+    def drop_application_before_upgrade(cascade: bool = False):
+        app.drop_application_before_upgrade(
+            console=console or cc,
+            app_name=app_model.fqn.identifier,
+            app_role=app_model.meta.role,
+            policy=policy,
+            is_interactive=is_interactive,
+            cascade=cascade,
+        )
+
+    pkg.action_bundle(action_ctx=ActionContext(get_entity=lambda *args: None))
+
+    return app.create_or_upgrade_app(
+        package_model=pkg_model,
+        stage_fqn=stage_fqn,
+        install_method=install_method,
+        drop_application_before_upgrade=drop_application_before_upgrade,
+    )
+
+
+def _setup_project(
+    app_pkg_role="package_role",
+    app_pkg_warehouse="pkg_warehouse",
+    app_role="app_role",
+    app_warehouse="app_warehouse",
+    setup_sql_contents="CREATE OR ALTER VERSIONED SCHEMA core;",
```

Comment:
does this need to be a parameter? the setup script is never checked or executed in unit tests; we can just hardcode it to an empty string (the file might not even need to exist)

Reply:
will fix it, not needed.